### ISSUE TITLE
Padding: Error Cannot convert value "10 12" to "Microsoft.Maui.Thickness"

### ISSUE THREAD

**Issue by SoyDiego** [dotnet/maui#15852]
### Description

Hi, I have a problem when I'm trying to add Padding to my `DataTrigger`.

I noticed two things:

- If I set only one value, for example: `<Setter Property="Padding" Value="10" />` works perfectly and compile and If I'm runtime and I set `<Setter Property="Padding" Value="10 12" />` or `<Setter Property="Padding" Value="10 12 10 12" />` works perfectly.
- But if my project is not in runtime and I'm trying to compile or build with <Setter Property="Padding" Value="10 12 10 12" />` , I get the error.

![image](https://github.com/dotnet/maui/assets/19764334/f5cec476-9763-4197-aadc-63c8820471e5)

### Steps to Reproduce

- If I set only one value, for example: `<Setter Property="Padding" Value="10" />` works perfectly and compile and If I'm runtime and I set `<Setter Property="Padding" Value="10 12" />` or `<Setter Property="Padding" Value="10 12 10 12" />` works perfectly.
- But if my project is not in runtime and I'm trying to compile or build with <Setter Property="Padding" Value="10 12 10 12" />` , I get the error.

### Link to public reproduction project repository

https://www.Idonthave.com

### Version with bug

7.0.86

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows 10

**Comment by jfversluis** [comment]
Does it work if you put a comma between the values?

**Comment by SoyDiego** [comment]
> Does it work if you put a comma between the values?

Thanks for your reply @jfversluis. It works with comma but my value before was `<Setter Property="Padding" Value="10 12" />` but now I need to do like this If I want to get the design that I want: <Setter Property="Padding" Value="13,10" />. Really, very weird!

Thanks and I close the issue! :)

**Comment by samhouts** [comment]
Yeah, it's pretty weird behavior that this works when Hot Reloading but then fails the build when you restart. @StephaneDelcroix seems like an easy fix?

**Comment by StephaneDelcroix** [comment]
tl;dr: input is wrong, you get an error

Not weird at all. Grab a blanket and a hot beverage, I'll tell you a story.

Once upon a time, there was those 2 converters.

Actually not in the mood for stories, but keep the hot beverage.

XAML values are supposed to be comma separated, and the syntax for thickness one of those three

- "u" => uniform value
- "h, v" => horizontal, vertical
- "l, t, r, b" => left, top, right, bottom

those are the values accepted by the type converter built in the compiler, and as "10 12" doesn't fit into any pattern, you get a compilation error.

The reason why HotReload accepts whitespace-separated values is that it reuse the runtime thicknesstypeconverter which also accepts CSS values, with the following format

- "u" => uniform value
- "v h" => vertical, horizontal
- "t h b" => top, horizontal, bottom
- "t r b l" => left, top, right, bottom

read this again, look closely at the order of arguments, and start reconsidering all of your career choices. There are plenty of good options, mine are "goat farming in Larzac", "crafting sailboats at the highest possible location", "growing a proper beard and play the guitar in the street" (I've tried long hair already, doesn't suits me).

The bug is not "can not convert "10 12" to thickness", the actual bug is that HotReload (and LoadFromXaml, and other places) accepts both syntax

**Comment by samhouts** [comment]
But Stephane! I'm a web developer! I love "v h"!

And yes, I did smack myself in the forehead when I added the commas and the margins flipped. 

Thanks for the additional context. :)
